### PR TITLE
Adding UUID related methods to identify Template groups & Rule templates

### PR DESCRIPTION
**Pull request by senthuran16**

### CODE REVIEW

File: components/org.wso2.carbon.business.rules.core/pom.xml [hunk 1]
```diff
@@ -79,6 +79,13 @@
             <artifactId>slf4j-log4j12</artifactId>
         </dependency>
 
+        <!-- GSON dependencies-->
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
+            <version>2.6.2</version>
```
Comment (ramindu90) [inline components/org.wso2.carbon.business.rules.core/pom.xml]: The version should go to the parent pom. Here, it should be only group id and artifact id

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/services/TemplateManagerService.java [hunk 2]
```diff
@@ -55,24 +58,38 @@ public TemplateManagerService() {
...
-            // Deploy templates, only if saving Business Rule definition is successful
-            // todo: (Q) is this ok?
+            saveBusinessRuleDefinition(businessRuleUUID, businessRuleFromTemplate); // todo : Implement method
+            // Deploy all derived templates, only if saving Business Rule definition is successful
             for (String templateUUID : derivedTemplates.keySet()) {
-                deployTemplate(templateUUID, derivedTemplates.get(templateUUID));
+                try {
+                    deployTemplate(templateUUID, derivedTemplates.get(templateUUID));
+                } catch (TemplateManagerException e) {
+                    log.error(e.getMessage()); // Exception is thrown from the above method itself
```
Comment (ramindu90) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/services/TemplateManagerService.java]: Usually, we use log.error(e.getMessage(), e). So it prints the stack trace as well

File: components/org.wso2.carbon.business.rules.core/pom.xml (hunk 1 above)
Comment (senthuran16) [reply components/org.wso2.carbon.business.rules.core/pom.xml]: Changed

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/services/TemplateManagerService.java [hunk 3]
```diff
@@ -55,24 +58,38 @@ public TemplateManagerService() {
...
         }
     }
 
     public void editBusinessRuleFromTemplate(String uuid, BusinessRuleFromTemplate businessRuleFromTemplate) { // todo: verify next lower level
-        Map<String, Template> derivedTemplates = deriveTemplates(businessRuleFromTemplate);
+        Map<String, Template> derivedTemplates = null;
+        try {
+            derivedTemplates = deriveTemplates(businessRuleFromTemplate);
+        } catch (TemplateManagerException e) {
+            e.printStackTrace(); // todo: implement
```
Comment (ramindu90) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/services/TemplateManagerService.java]: I think you have mistakenly put "printStackTrace"

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/services/TemplateManagerService.java [hunk 4]
```diff
@@ -406,7 +426,10 @@ public RuleTemplate getRuleTemplate(BusinessRuleFromTemplate businessRuleFromTem
      * @throws TemplateManagerException
      */
     public void saveBusinessRuleDefinition(String uuid, BusinessRuleFromTemplate businessRuleFromTemplate) throws TemplateManagerException {
-        //todo: implement
+        System.out.println("[SAVED BUSINESS RULE DEFINITION]__________");
```
Comment (ramindu90) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/services/TemplateManagerService.java]: Remove there sout's if possible

**Pull request merged**